tests/drivers/gpio/gpio_basic_api/src/test_gpio_port.c

```diff
@@ -245,6 +245,12 @@ static int pin_physical(void)
 	rc = gpio_pin_set_raw(dev, PIN_OUT, 32);
 	zassert_equal(rc, 0,
 		      "pin_set_raw high failed");
+
+	/* ensure that gpio configuration flags do reflect */
+	rc = gpio_pin_get_raw(dev, PIN_IN);
```

Comment:
Why isn't this covered with the existing following test that reads the raw pin value and verifies it's true?

Reply:
I'm trying to solve issue described there https://github.com/zephyrproject-rtos/zephyr/issues/25148

Reply:
Interesting: apparently the MCHP-XEC GPIO is a bit unusual.  The purpose of initialize-to-level is to ensure there's no transient driving of the signal at the wrong level, which requires that the outputs be configured before the pin direction is set.  The fix in #25103 shouldn't have been necessary because that's what the code did.  So this peripheral doesn't record the output settings unless the direction is already configured as output?  This must have something to do with the "alternate output disable" feature, but I would expect that to be a problem only on the first configuration.

In any case, I don't believe this fix would have detected the problem because as I note the existing test at this point in the code:
```
zassert_equal(raw_in(), true,
              "pin_set_raw high failed");
```
should be functionally equivalent to the one that was added.

I'm not clear on how things failed, and I don't have the hardware so I can't reproduce the original problem by reverting #25103.  Even the `setup()` function confirms that physical high and low work correctly.  But given that https://github.com/zephyrproject-rtos/zephyr/issues/25101#issue-614248346 focuses on a test of the interrupt capability it's possible there's a problem specific to the immediately preceding operations.  I would expect anything that would detect the problem to have to be done there; by the time that test is reached we've already verified that high and low physical and logical work correctly.

I'm not comfortable approving this unless there's more evidence it fixes a problem.  Looping in @mnkp who may be able to see what's going on.